1.8/usage/tutorials/tutorial/service-discovery.md

```diff
@@ -0,0 +1,71 @@
+In this section, we will learn about DC/OS service discovery by exploring the different options for service discovery for apps in DC/OS.
+
+# Steps
+  [Service discovery](https://dcos.io/docs/1.8/usage/service-discovery/) allows you to connect to your applications without necessarily knowing where they are running. Service discovery is particularly useful in cases where applications may fail and be restarted on a different host.
+
+  DC/OS provides two options for service discovery: Mesos-DNS and Named virtual IPs.
+  * SSH into your cluster to see how these service discovery methods work: `dcos node ssh --master-proxy --leader`,
+  * [Mesos-DNS](https://dcos.io/docs/1.8/usage/service-discovery/mesos-dns/) assigns a Mesos-DNS for every Marathon app. The naming pattern is  *task.scheduler.mesos* and the default scheduler for jobs is `marathon`, so the Mesos-DNS name for our redis service is *redis.marathon.mesos*.
+
+  Let us use [dig](https://en.wikipedia.org/wiki/Dig_(command)) to retrieve the address record (also called the A record): `dig redis.marathon.mesos`.
+
+  The answer should contain similar to this response:
+
+  ~~~~
+  ;; ANSWER SECTION:
+  redis.marathon.mesos. 60  IN  A 10.0.0.43
+  ~~~~
+
+  The response tells us that the host for the `redis.marathon.mesos` is 10.0.0.43.
+
+  The A record only contains information about the host. In order to connect to the service, we also need to know the port. Use the following dig command to access the Service locator (SRV) DNS record, which also provides port information: `dig srv _redis._tcp.marathon.mesos`.
+  The answer should look similar to this response:
+
+  ~~~~
+  ;; ANSWER SECTION:
+  _redis._tcp.marathon.mesos. 60  IN  SRV 0 0 30585 redis-1y1hj-s1.marathon.mesos.
+
+  ;; ADDITIONAL SECTION:
+  redis-1y1hj-s1.marathon.mesos. 60 IN  A 10.0.0.43
+  ~~~~
+
+   So we now know that our redis app is running on 10.0.0.43:30585.
+
+    * [Named Vips](https://dcos.io/docs/1.8/usage/service-discovery/load-balancing-vips/) allow you to assign name/port pairs to your apps. Named VIPs allow you to assign meaningful names to your apps.
+    For example, we can assign a named VIP to our Redis service by adding the following to the package definition:
+
+    ~~~
+    "VIP_0": "/redis:6379"
+    ~~~
+
+    The full name is then generated using the following schema:
+    *vip-name.scheduler.l4lb.thisdcos.directory:vip-port*. In the example above, you can access the Redis service from within the cluster at the following address: redis.marathon.l4lb.thisdcos.directory:6379.
+
+# Outcome
+You know how to use service discovery to connect to your application from within your DC/OS cluster.
+
+# Deep Dive
+What are the differences between Mesos-DNS and Named VIPs?
+
+## Mesos-DNS
+Mesos-DNS is a rather simple solution to find applications inside the cluster and DNS is supported by many applications.
```

Comment:
Suggest: "Mesos-DNS is a rather simple solution to finding applications inside the cluster. While DNS is supported by many applications, Mesos-DNS has the following drawbacks:"